tests/test_quantization_metric.py

```diff
@@ -0,0 +1,192 @@
+from nncf.quantization_metric import NetworkQuantizationShareMetric, MemoryСostMetric, ShareBlueEdgesCompressedGraph
+from nncf import create_compressed_model, NNCFConfig
+#from nncf.quantization.quantizer_propagation import DEFAULT_QUANT_TRAIT_TO_OP_DICT, QuantizationTrait
+from tests import test_models
+
+def get_basic_quantization_config():
+    config = NNCFConfig()
+    config.update({
+        "model": "AlexNet",
+        "input_info":
+            {
+                "sample_size": [1, 3, 32, 32],
+            },
+        "compression":
+            {
+                "algorithm": "quantization",
+                "initializer": {
+                    "range": {
+                        "num_init_steps": 0
+                    }
+                }
+            }
+    })
+
+    return config
+
+TestStruct = namedtuple('TestStruct',
+                        ('initializers',
+                         'activations',
+                         'weights',
+                         'ignored_scopes',
+                         'table'))
+
+NETWORK_QUANTIZATION_SHARE_METRIC_TEST_CASES = [
+    TestStruct(
+        initializers={"precision": {
+            "bitwidth_per_scope":
+            [[2, 'AlexNet/Sequential[features]/NNCFConv2d[0]'],
+             [4, 'AlexNet/Sequential[features]/NNCFConv2d[6]']]
+            }},
+        activations={
+            "mode": "asymmetric",
+            "bits": 8,
+            "signed": True,
+            },
+        weights={
+            "mode": "symmetric",
+            "per_channel": True,
+            "bits": 8
+            },
+        ignored_scopes=['AlexNet/Sequential[classifier]'],
+        table=[['signed', '0', '26.087', '0', '26.087', '26.087'],
+               ['2', '4.348', '4.348', '4.348', '4.348', '8.696'],
+               ['4', '4.348', '0', '4.348', '0', '4.348'],
+               ['symmetric', '21.739', '0', '21.739', '0', '21.739'],
+               ['8', '13.043', '21.739', '13.043', '21.739', '34.783'],
+               ['per_channel', '21.739', '0', '21.739', '0', '21.739'],
+               ['assymmetric', '0', '26.087', '0', '26.087', '26.087']]),
+    TestStruct(
+        initializers={"precision": {
+            "bitwidth_per_scope":
+            [[2, 'AlexNet/Sequential[features]/NNCFConv2d[0]'],
+             [4, 'AlexNet/Sequential[features]/NNCFConv2d[6]']]
+        }},
+        activations={"bits": 8},
+        weights={"bits":6},
+        ignored_scopes=['AlexNet/Sequential[classifier]'],
+        table=[['signed', '0', '0', '0', '0', '0'],
+               ['2', '4.348', '4.348', '4.348', '4.348', '8.696'],
+               ['4', '4.348', '0', '4.348', '0', '4.348'],
+               ['symmetric', '21.739', '26.087', '21.739', '26.087', '47.826'],
+               ['6', '13.043', '0', '13.043', '0', '13.043'],
+               ['8', '0', '21.739', '0', '21.739', '21.739'],
+               ['per_channel', '0', '0', '0', '0', '0'],
+               ['assymmetric', '0', '0', '0', '0', '0']]),
```

Comment:
It is completely unintelligible from this specificaton which entry corresponds to what. Anyway, you shouldn't compare the table rows, you should compare each separate metric.

Reply:
This still is not fixed.

Reply:
This is STILL not fixed. You are still comparing the rows.